### PR TITLE
yarprobotstatepublisher

### PR DESCRIPTION
**Pull request by traversaro**
YARP and iDynTree powered version of ros `robot_state_publisher` . 

Main advantages over ROS's `robot_state_publisher` : 
* You can specify the desired root of the tree over all the frames of the robot 
* Also the sensor frames are broadcasted (thanks to the [`addSensorFramesAsAdditionalFrames`](http://wiki.icub.org/codyco/dox/html/idyntree/html/structiDynTree_1_1URDFParserOptions.html#a0b4a5be086b3511d0e582c91c60697f4) option of iDynTree's URDF parser ). 
* Uses YARP's `tfClient` device to publish the tf transforms, and uses YARP-ROS compatibility to read the joint states from the `/joint_states` topic, so it can run easily on Windows even without ROS. 

cc @randaz81 @aerydna

### CODE REVIEW

**Comment by aerydna** [comment]
It seems Christmas is arrived very early this year..

File: src/tools/yarprobotstatepublisher/src/robotstatepublisher.cpp [hunk 1]
```diff
@@ -0,0 +1,236 @@
...
+{
+}
+
+
+/************************************************************/
+bool YARPRobotStatePublisherModule::configure(ResourceFinder &rf)
+{
+    string name="yarprobotstatepublisher";
+    m_rosNode = new yarp::os::Node("/yarprobotstatepublisher");
+    string robot=rf.check("robot",Value("isaacSim")).asString();
```
Comment (traversaro) [inline src/tools/yarprobotstatepublisher/src/robotstatepublisher.cpp]: @aerydna ops

**Comment by traversaro** [comment]
@barbalberto I rebased the branch on the latest devel. Compilation of iDynTree should be straightforward, just donwload it and compile it as any CMake project (perhaps pay attention to select `IDYNTREE_USES_YARP` to make sure to compile it with YARP support). 
I actually was wrong, the joint state is read from the `/joint_states` topic as the classical ROS's `robot-state-publisher` .

File: src/tools/yarprobotstatepublisher/src/robotstatepublisher.cpp [hunk 2]
```diff
@@ -162,7 +162,8 @@ bool YARPRobotStatePublisherModule::close()
 
     m_baseFrameIndex = iDynTree::FRAME_INVALID_INDEX;
 
-    delete m_rosNode;
+    if(m_rosNode)
+        delete m_rosNode;
```
Comment (traversaro) [inline src/tools/yarprobotstatepublisher/src/robotstatepublisher.cpp]: Good catch. Would it make sense to also add `m_rosNode = nullptr`?

**Comment by barbalberto** [comment]
Succesfully tested with Gazebo

**Comment by diegoferigo** [comment]
@lucaTagliapietra This coupled with `tfServer` might avoid us using `roslaunch`, let's remember this :tada:

**Pull request merged**